edxval/models.py

```diff
@@ -612,6 +612,55 @@ def __unicode__(self):
         return u'{course_id} - {provider}'.format(course_id=self.course_id, provider=self.provider)
 
 
+class ThirdPartyTranscriptCredentialsState(models.Model):
+    """
+    State of transcript credentials for a course organization
+    """
+    class Meta:
+        unique_together = ('org', 'provider')
+
+    org = models.CharField(verbose_name='Course Organization', max_length=32)
+    provider = models.CharField(
+        verbose_name='Transcript Provider',
+        max_length=20,
+        choices=TranscriptProviderType.CHOICES,
+    )
+    exists = models.BooleanField(default=False, help_text='Transcript credentials state')
+
+    @classmethod
+    def update_or_create(cls, org, provider, exists):
+        """
+        Update or create credentials state.
+        """
+        instance, created = cls.objects.update_or_create(
+            org=org, provider=provider,
+            defaults={'exists': exists},
+        )
+
+        return instance, created
+
+    def __unicode__(self):
+        """
+        Returns unicode representation of provider credentials state for an organization.
+
+        NOTE: Message will look like below:
+            edX has Cielo24 credentials
+            edX has no 3PlayMedia credentials
+        """
+        return u'{org} {state} {provider} credentials'.format(
+            org=self.org, provider=self.provider, state='has' if self.exists else "doesn't have"
+        )
+
+
+@receiver(models.signals.pre_save, sender=ThirdPartyTranscriptCredentialsState)
+def validate_provider(sender, instance, **kwargs):  # pylint: disable=unused-argument
+    """
+    Validate that input provider value must be one of TranscriptProviderType.CHOICES.
+    """
+    if instance.provider not in dict(TranscriptProviderType.CHOICES).keys():
+        raise ValidationError('Invalid transcript provider value')
```

Comment:
@muhammad-ammar I think its alright to skip the this signal to validate the data. We can validate it in api utils which are responsible for creating an actual state record in db, Or even we may not validate as its getting validated from the view which is calling val's api.